misc/inforequests.md

```diff
@@ -0,0 +1,767 @@
+![](assets/infore)
+
+## `Inforequest Draft`
+
+Relations:
+* `applicant`: User; May NOT be NULL.\
+  The draft owner, the future inforequest applicant.
+* `obligee`: Obligee; May be NULL.\
+  The obligee the inforequest will be sent to, if the user has already set it.
+* `attachment_set`: List of Attachments; May be empty; Ordered by id.\
+  Attachments for the future request action.
+
+Properties:
+* `subject`: String; May be empty.\
+  Draft subject for the future request action.
+* `content`: String; May be empty.\
+  Draft content for the future request action.
+
+## `Inforequest`
+
+Represents a single inforequest. Inforequests are ordered by their submission date by default.
+
+Relations:
+* `applicant`: User; May NOT be NULL.\
+  The inforequest owner, the user who submitted it.
+* `email_set`: List of E-mail Messages; through Inforequest E-mails; May be empty; Ordered by
+  processed date.\
+  List of all inbound and outbound messages related to the inforequest, decided or undecided. The
+  `unique_email` address generated for the inforequest is used for communication with all obligees
+  the request was advanced to. It is up to the applicant ~~and/or some heuristics~~ to decide to
+  which branch the action contained in the e‑mail belongs. The e‑mails are ordered by the date we
+  sent or received them, respectively. Queued outbound e-mails waiting to be sent are ordered by
+  their id.
+* `branch_set`: List of Branches; May NOT be empty; Ordered by obligee name.\
+  List of all branches related to the inforequest. Every inforequest contains at least one branch,
+  the main branch.
+* `actiondraft_set`: List of Action Drafts; May be empty; Ordered by id.\
+  List of all action drafts the applicant created for this inforequest. The inforequest may contain
+  at most one draft of each action type.
+* `inforequestemail_set`: List of Inforequest E-mails; May be empty; Ordered by id.
+
+Computed Relations:
+* `branch`: Branch; May NOT be NULL; Read-only.\
+  Main branch of communication with the original obligee the inforequest was submitted to. The
+  inforequest must contain exactly one main branch. If the inforequest was advanced to other
+  obligees, the respective advancement action contains recursive sub-branches of communication with
+  new obligees.
+* `undecided_set`: List of E-mail Messages; May be empty; Read-only.\
+  List of undecided inbound messages waiting for user decision.
+* `oldest_undecided_email` and `newest_undecided_email`: E-mail Message; May be NULL; Read-only.
+
+Properties:
+* `applicant_name`, `applicant_street`, `applicant_city` and `applicant_zip`: String; May NOT be
+  empty.\
+  To freeze applicant current contact information for the case he changes it in the future. The
+information is frozen to its state at the moment the inforequest was submitted.
+* `unique_email`: E-mail; May NOT be empty; Unique.\
+  Every inforequest has a unique e‑mail address of the form *@mail.chcemvediet.sk. This address is
```

Comment:
Hviezdicku v MD treba escapnut, lebo potom to domota formatovanie a vsetok text za nou je kurizovou.

Reply:
:ok: